pkg/reverse_hash/factory.go

```diff
@@ -85,7 +85,7 @@ func (f *Factory) initOnChainRHSCli(ctx context.Context, kmsKey *kms.KeyID) (pro
 	if err != nil {
 		return nil, err
 	}
-	cli, err := proofEth.NewReverseHashCli(f.onChainTreeStoreContract, f.ethClient.GetEthereumClient(), txOpts, f.responseTimeout)
+	cli, err := proofEth.NewReverseHashCli(f.ethClient.GetEthereumClient(), f.onChainTreeStoreContract, txOpts.From, txOpts.Signer)
```

Comment:
@volodymyr-basiuk is it an only thing to change to support new version of onchain rhs cli?
@martinsaporiti please double check that onchain rhs is working on staging / test

Reply:
I have checked with `ISSUER_CREDENTIAL_STATUS_RHS_MODE=OnChain`, it works